### DISCUSSION TITLE
HOW TO - EdgeVPN installation on LXD System containers

### DISCUSSION THREAD

**Discussion by bmullan** [mudler/edgevpn#49] [Ideas]
I just installed EdgeVPN in LXD containers running on 3 HOST Servers and 1 here at my home.

On each LXD HOST Server node I created 1 LXD Container.

Normally, LXD containers (by default) are isolated on each LXD HOST on a 10.x.x.x network.   They have access to their HOST and the Internet.  But other local/remote servers cannot directly connect (re w ssh etc) to another Server's LXD containers.

So EdgeVPN inter-connected ALL of the local/remote LXD Containers on a 172.16.0.0/12 Network VPN.  Applications in 1 LXD Container can now interact w a Remote HOST's LXD Container's Ubuntu OS -or- applications running in the other LXD "system" container..!

Works great!  

FYI all my LXD containers were Ubuntu 22.04 w Gnome Desktop & xrdp installed so I can use xfreerdp to get a Remote Desktop on any of the local/remote LXD Containers when needed.

If anyone is interested I can polish up my Configuration Notes a bit and post them.   

To be honest the hardest part of doing this was the EdgeVPN documentation.   It left many things to "j_ust figure out_" because the Doc's in places were just not detailed/specific  enough to understand at first.

For those unfamiliar with LXD containers.   Unlike Docker containers which are considered "application" containers, LXD containers are termed "system" containers because they can run a full Linux OS w/systemd, etc.   Each LXD Container uses it's HOST's linux kernel.   

LXD containers can be created with many different Linux OS's including:
- Debian
- Fedora
- CentOS
- Ubuntu
- Alpine
- Suse
- etc etc

LXD also now supports QEMU based VM's using essentially the same CLI command structure as used for the LXD Containers.

Reference:   **[linuxcontainers.org LXD Introduction](https://linuxcontainers.org/lxd/introduction/)**

Brian

**Comment by mudler** [comment]
Hey :wave: @bmullan deeply apologize for the late reply. 

This is looking just awesome, please feel free to polish the configuration notes and post them here - I'd be more than happy actually to have them in the official docs if you feel to take a stab at it.

**Reply by bmullan** [reply]
@mudler Sorry for delay in responding but I was sick for several days.
I'll try to write something up and email it to you or whatever for review/comments.
Brian
